main/acle.md

```diff
@@ -4836,6 +4838,47 @@ if such a restoration is necessary. For example:
    }
 ```
 
+## `__arm_agnostic`
+
+A function with the `__arm_agnostic` [keyword attribute](#keyword-attributes) must
+preserve the architectural state that is specified by its arguments when such
+state exists at runtime. The function is otherwise unconcerned with the contents
+of this state.
+
+The `__arm_agnostic` [keyword attribute](#keyword-attributes) applies to
+**function types** and accepts the following arguments:
+
+```"sme_za_state"```
+
+*   If the function is defined and PSTATE.ZA is available, the definition must
+    preserve all architectural state enabled by PSTATE.ZA.
+
+    It is the compiler's responsibility to ensure that this state is preserved.
+
+    The compiled function must be forward-compatible and should not make any
+    assumptions on what state is enabled by PSTATE.ZA, which could be done by
+    relying on ABI routines available on the platform for the allocation of
+    a buffer and the saving and restoring of state.
+
+*   If PSTATE.ZA is available, then the [abstract machine](#abstract-machine)
+    ensures that on return from the function, the value of PSTATE.ZA is the same
+    as it was on entry to the function.
+
+*   If the function forms part of the object code's ABI, that object code
+    function has a “ZA-compatible interface”; see [[AAPCS64]](#AAPCS64)
+    for more details.
+
+*   It is not valid for a function declaration with `__arm_agnostic("sme_za_state")`
```

Comment:
`__arm_new` is only meaningful for definitions, not declarations.  I'm not sure it should be a semantic requirement, since:
```
__arm_new("za") void foo(void) …stuff… {
    …
}
```
isn't semantically different from:
```
__arm_new("za") void foo1(void) {
    …
}
void foo(void) …stuff… {
    foo1();
}
```
(It's reasonable for an implementation to say that it doesn't support `__arm_new` for now though.  This would be the equivalent of a GCC “sorry” diagnostic, but it'd be an implementation choice rather than something that the ACLE should document.)

I think we can then say:

> `It is not valid … to [share](#shares-state) PSTATE.ZA state with its caller.`